src/util/expr.h

```diff
@@ -85,9 +85,8 @@ class exprt:public irept
     return static_cast<const typet &>(find(ID_type));
   }
 
-  /// Amount of nodes this expression tree contains. This is the size of the
-  /// actual tree, ignoring memory/sub-tree sharing.
```

Comment:
I'd keep this comment. This refers what @martin-cs meant by structural sharing.

Reply:
Added in.